api/v1beta1/kafkacluster_types.go

```diff
@@ -225,20 +225,23 @@ type AlertManagerConfig struct {
 
 // ExternalListenerConfig defines the external listener config for Kafka
 type ExternalListenerConfig struct {
-	Type                 string `json:"type"`
-	Name                 string `json:"name"`
+	*CommonListenerSpec  `json:",inline"`
 	ExternalStartingPort int32  `json:"externalStartingPort"`
-	ContainerPort        int32  `json:"containerPort"`
 	HostnameOverride     string `json:"hostnameOverride,omitempty"`
 }
 
 // InternalListenerConfig defines the internal listener config for Kafka
 type InternalListenerConfig struct {
-	Type                            string `json:"type"`
-	Name                            string `json:"name"`
-	UsedForInnerBrokerCommunication bool   `json:"usedForInnerBrokerCommunication"`
-	ContainerPort                   int32  `json:"containerPort"`
-	UsedForControllerCommunication  bool   `json:"usedForControllerCommunication,omitempty"`
+	*CommonListenerSpec             `json:",inline"`
```

Comment:
Is `CommonListenerSpec` optional? If not than it's no worth making it a pointer.